pkg/generators/openapi_test.go

```diff
@@ -495,3 +502,62 @@ Dependencies: []string{
 },
 `, buffer.String())
 }
+
+func TestExtensions(t *testing.T) {
+	err, assert, buffer := testOpenAPITypeWritter(t, `
+package foo
+
+// Blah is a test.
+// +k8s:openapi-gen=true
+// +k8s:openapi-gen=x-kubernetes-type-tag:type_test
+type Blah struct {
+	// a member with a list type
+	// +listType=map
+	// +listMapKey=port
+	// +listMapKey=protocol
```

Comment:
Why not `listMapKeys=port,protocol`?

Reply:
We would have to make sure the separator character is not in any value. If we want to change it, let's do it in another small, focused PR.

Reply:
Commas are not going to be in any field names?

Reply:
I actually like this form, because it doesn't make it obvious to people that they can have multiple keys (which we certainly don't want to encourage)